### PR TITLE
Store and sync AGS claim

### PR DESCRIPTION
**Pull request by michaelwheeler**
This is intended as the first of multiple PRs implementing AGS support. These changes simply store the AGS endpoint and permissions on the launch context.

### CODE REVIEW

File: lti_tool/utils.py [hunk 1]
```diff
@@ -138,44 +138,53 @@ def sync_user_from_launch(lti_launch: LtiLaunch) -> LtiUser:
...
+        "lineitems_url": ags_claim.get("lineitems", ""),
+        "can_query_lineitems": AgsScope.QUERY_LINEITEMS in ags_scope,
+        "can_manage_lineitems": AgsScope.MANAGE_LINEITEMS in ags_scope,
+        "can_publish_scores": AgsScope.PUBLISH_SCORES in ags_scope,
+        "can_access_results": AgsScope.ACCESS_RESULTS in ags_scope,
+    }
+
+
 def sync_context_from_launch(lti_launch: LtiLaunch) -> LtiContext:
     context_claim = lti_launch.context_claim
```
Comment (doreilly) [inline lti_tool/utils.py:156]: This line looks like its obsolete at this point. I'm guessing it's ok to remove?

File: lti_tool/utils.py [hunk 2]
```diff
@@ -138,44 +138,53 @@ def sync_user_from_launch(lti_launch: LtiLaunch) -> LtiUser:
...
-        }
-        if nrps_endpoint:
-            defaults["memberships_url"] = nrps_endpoint
-        context, _created = LtiContext.objects.update_or_create(
-            deployment=lti_launch.deployment,
-            id_on_platform=context_claim["id"],
-            defaults=defaults,
-        )
+    context_types = context_claim.get("type", [])
+    defaults = {
```
Comment (doreilly) [inline lti_tool/utils.py:161]: I like the re-arrangement of the context's update/creation here.

**Review by doreilly — COMMENTED** [review]
Other than the one errant line that should likely be removed, I think this is plenty good for merge.

File: lti_tool/utils.py (hunk 1 above)
Comment (michaelwheeler) [reply lti_tool/utils.py]: Yep, thanks for taking a close look!

**Pull request merged**